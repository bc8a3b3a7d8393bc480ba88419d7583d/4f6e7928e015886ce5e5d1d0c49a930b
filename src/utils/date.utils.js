import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
