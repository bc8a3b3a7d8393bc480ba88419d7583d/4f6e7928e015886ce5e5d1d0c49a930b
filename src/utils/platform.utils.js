import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
