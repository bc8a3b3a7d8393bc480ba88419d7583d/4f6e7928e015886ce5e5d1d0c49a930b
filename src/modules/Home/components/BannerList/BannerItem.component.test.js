import { testSnapshots } from '../../../../utils/test.utils';

import Component from './BannerItem.component';

describe('Modules/Home - BannerItem', () => {
  testSnapshots(Component, [
    {
      description: 'basic render',
      props: {
        onPress: jest.fn(),
        title: 'Flights',
        image: 'flight',
      },
    },
  ]);
});
