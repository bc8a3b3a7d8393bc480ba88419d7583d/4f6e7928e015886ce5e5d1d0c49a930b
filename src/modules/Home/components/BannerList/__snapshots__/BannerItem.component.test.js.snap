// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Home - BannerItem check render grid item 1`] = `
<TouchableWithoutFeedback
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 5,
          "marginBottom": 10,
          "marginLeft": 10,
          "overflow": "hidden",
        },
        Object {
          "marginRight": 10,
        },
        false,
        Object {},
      ]
    }
  >
    <ImageBackground
      resizeMode="cover"
      source={
        Object {
          "uri": "https://tvlk.imgix.net/imageResource/2018/11/29/1543491168198-a6155734ddedef7ec6c4c230e5386bb5.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#8f8f8f",
            "borderRadius": 5,
            "padding": 10,
          },
          Object {
            "height": 236.66666666666666,
            "width": 236.66666666666666,
          },
        ]
      }
    />
  </Component>
</TouchableWithoutFeedback>
`;

exports[`Modules/Home - BannerItem snapshots basic render 1`] = `
<TouchableWithoutFeedback
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 5,
          "marginBottom": 10,
          "marginLeft": 10,
          "overflow": "hidden",
        },
        Object {
          "marginRight": 10,
        },
        false,
        Object {},
      ]
    }
  >
    <ImageBackground
      resizeMode="cover"
      source={
        Object {
          "uri": "https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#8f8f8f",
            "borderRadius": 5,
            "padding": 10,
          },
          Object {
            "height": 540.7407407407408,
            "width": 540.7407407407408,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://tvlk.imgix.net/imageResource/2018/03/28/1522238393788-a1ea3b1b31fe85fe7e8d000f9e3c0ff3.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "bottom": 10,
              "color": "white",
              "fontFamily": "Oswald-Regular",
              "fontSize": 12,
              "left": 10,
              "position": "absolute",
              "right": 10,
            },
            Object {
              "color": "#1b1b1b",
            },
          ]
        }
      >
        See All Promos
      </Text>
      <Component
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="playcircleo"
          size={40}
        />
      </Component>
    </ImageBackground>
  </Component>
</TouchableWithoutFeedback>
`;

exports[`Modules/Home - BannerItem snapshots default number of column is 1 1`] = `
<TouchableWithoutFeedback
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 5,
          "marginBottom": 10,
          "marginLeft": 10,
          "overflow": "hidden",
        },
        Object {
          "marginRight": 10,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "paddingHorizontal": 10,
        },
        Object {},
      ]
    }
  >
    <ImageBackground
      resizeMode="cover"
      source={
        Object {
          "uri": "https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#8f8f8f",
            "borderRadius": 5,
            "padding": 10,
          },
          Object {
            "height": 540.7407407407408,
            "width": 540.7407407407408,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://tvlk.imgix.net/imageResource/2018/03/28/1522238393788-a1ea3b1b31fe85fe7e8d000f9e3c0ff3.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "bottom": 10,
              "color": "white",
              "fontFamily": "Oswald-Regular",
              "fontSize": 12,
              "left": 10,
              "position": "absolute",
              "right": 10,
            },
            Object {
              "color": "#1b1b1b",
            },
          ]
        }
      >
        See All Promos
      </Text>
      <Component
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="playcircleo"
          size={40}
        />
      </Component>
    </ImageBackground>
  </Component>
</TouchableWithoutFeedback>
`;

exports[`Modules/Home - BannerItem snapshots should render full banner section 1`] = `
<TouchableWithoutFeedback
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 5,
          "marginBottom": 10,
          "marginLeft": 10,
          "overflow": "hidden",
        },
        Object {
          "marginRight": 10,
        },
        false,
        Object {},
      ]
    }
  >
    <ImageBackground
      resizeMode="cover"
      source={
        Object {
          "uri": "https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#8f8f8f",
            "borderRadius": 5,
            "padding": 10,
          },
          Object {
            "height": 973.3333333333334,
            "width": 973.3333333333334,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://tvlk.imgix.net/imageResource/2018/03/28/1522238393788-a1ea3b1b31fe85fe7e8d000f9e3c0ff3.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "bottom": 10,
              "color": "white",
              "fontFamily": "Oswald-Regular",
              "fontSize": 12,
              "left": 10,
              "position": "absolute",
              "right": 10,
            },
            Object {
              "color": "#1b1b1b",
            },
          ]
        }
      >
        See All Promos
      </Text>
      <Component
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="playcircleo"
          size={40}
        />
      </Component>
    </ImageBackground>
  </Component>
</TouchableWithoutFeedback>
`;
